arch/arm/core/irq_manage.c

```diff
@@ -84,19 +84,18 @@ int _arch_irq_is_enabled(unsigned int irq)
  */
 void _irq_priority_set(unsigned int irq, unsigned int prio, u32_t flags)
 {
-	/* Hardware priority levels 0 and 1 reserved for Kernel use.
-	 * So we add 2 to the requested priority level. If we support
-	 * ZLI, 2 is also reserved so we add 3.
+	/* Hardware priority levels can be reserved for Kernel use.
+	 * So we add to the requested priority level.
 	 */
 
 #if CONFIG_ZERO_LATENCY_IRQS
-	/* If we have zero latency interrupts, that makes priority level 2
+	/* If we have zero latency interrupts, the last priority level is
 	 * a case with special semantics; it is not masked by irq_lock().
 	 * Our policy is to express priority levels with special properties
 	 * via flags
 	 */
 	if (flags & IRQ_ZERO_LATENCY) {
-		prio = 2;
+		prio = _IRQ_PRIO_OFFSET - 1;
```

Comment:
so, _IRQ_PRIO_OFFSET is 2 (when we configure for zero-latency IRQs), so this changes the priority setting from 2 to 1. Why do we want to do that?

Reply:
After, doing the investigation, I think that this could be done a bit better. We should not require the Zero-latency IRQ level to be the _last_ reserved offset. Instead, we should have a #define for the _ZERO_LATENCY_IRQ_PRIO_OFFSET, as we have for _IRQ_PRIO_OFFSET.

Reply:
How about setting _EXC_ZERO_LATENCY_PRIO 1 in cortex_m/exc.h together with SVC/FAULT_PRIO?

Reply:
This is also an option, yes. Note that we might need to do some more refactoring when calculating the final OFFSET.